src/bitmap.rs

```diff
@@ -74,9 +79,15 @@ impl<const PAGE_SIZE: usize> PageAllocator for BitmapPageAllocator<PAGE_SIZE> {
     }
 
     fn dealloc_pages(&mut self, pos: usize, num_pages: usize) {
-        // TODO: not decrease `used_pages` if deallocation failed
-        self.used_pages -= num_pages;
-        self.inner.dealloc((pos - self.base) / PAGE_SIZE)
+        if match num_pages.cmp(&1) {
+            core::cmp::Ordering::Equal => self.inner.dealloc((pos - self.base) / PAGE_SIZE),
```

Comment:
can we replace it with `dealloc_contiguous((pos - self.base) / PAGE_SIZE, 1)`? 

then we can unify the `equal` and `greater` cases

Reply:
I'm just as curious as you about why it was written like this before.

Reply:
`alloc_contigous` has a longer code path than `alloc` and may perform worse.

Reply:
> `alloc_contiguous`/`dealloc_contiguous` has a longer code path than `alloc`/`dealloc` and may perform worse.

should we care for that, change it back to `dealloc`?

Reply:
You decide

Reply:
> You decide

fixed